File: src/java/EntityFX/src/EntityFX/Core/Generic/ParallelHashBenchmark.java
package EntityFX.Core.Generic;

import java.io.FileNotFoundException;
import java.util.concurrent.ExecutionException;

import EntityFX.Core.Writer;

public class ParallelHashBenchmark extends HashBase<BenchResult[]> {

    public ParallelHashBenchmark(Writer writer, boolean printToConsole) throws FileNotFoundException {
        super(writer, printToConsole);
        isParallel = true;
    }

    @Override
    public BenchResult[] benchImplementation() throws InterruptedException, ExecutionException {
        return benchInParallel(() -> 0, a -> {
            byte[] hash = new byte[0];
            for (int i = 0; i < Iterrations; i++) {
                hash = doHash(i, arrayOfBytes);
            }
            return hash;
        }, (a, r) -> {
        });
    }
}

File: src/python/entityfx/parallel_hash_benchmark.py

from entityfx.hash_base import HashBase
from entityfx.writer import Writer

class ParallelHashBenchmark(HashBase):  
    def __init__(self, writer: Writer, print_to_console : bool=True, is_enabled : bool=True) -> None:
        super().__init__(writer, print_to_console, is_enabled)
        self.is_parallel = True

    def benchImplementation(self) -> list:
        return self.bench_in_parallel(lambda : 0, lambda a: self.benchFunc(a), lambda a, r : self.setBenchResultFunc(a, r))

    def benchFunc(self, a):
        result = None
        for i in range(0, self._iterrations):
            result = HashBase._doHash(i, self._array_of_bytes)
        return result

    def setBenchResultFunc(self, a, r):
        pass